File: src/ssr_pkg/scripts/lidar_node.py
#!/usr/bin/env python
# encoding:utf8

import rospy
from std_msgs.msg import String

if __name__ == "__main__":
    rospy.init_node("lidar_node")
    rospy.logwarn("lidar!py!")

    pub = rospy.Publisher("my_lidar_topic", String, queue_size=10)
    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        rospy.loginfo("lidar!!!py")
        msg = String()
        msg.data = "msg_of_lidar_py"
        pub.publish(msg)
        rate.sleep()

File: src/ssr_pkg/src/lidar_node.cpp
#include <ros/ros.h>
#include<std_msgs/String.h>

int main(int argc, char  *argv[])
{
    ros::init(argc, argv, "lidar_node");
    ros::NodeHandle nh;
    ros::Publisher  pub =  nh.advertise<std_msgs::String>("my_lidar_topic", 10);
    ros::Rate loop_rate(10);

    while (ros::ok())
    {
        printf("lidar!!!\n");
        std_msgs::String msg;
        msg.data = "lidar_msg";
        pub.publish(msg);
        loop_rate.sleep();
    }

    return 0;
} 